### PR TITLE
[Hasura] Add Spec

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature
**What is the current behavior? (You can also link to an open issue here)**
#281

### CODE REVIEW

File: dev/hasura.ts [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+const adminSecretOption: Fig.Option = {
+  name: "--admin-secret",
+  description: "Admin secret for Hasura GraphQL engine",
+  args: stringArg,
+};
+
+const certificateAuthorityOption: Fig.Option = {
+  name: "--certificate-authority",
+  description: "Path to a cert file for the certificate authority",
+  args: stringArg,
```
Comment (QuiiBz) [inline dev/hasura.ts:54]: This can have a template of `filepaths`

File: dev/hasura.ts [hunk 2]
```diff
@@ -0,0 +1,732 @@
...
+            name: "string",
+            template: "filepaths",
+          },
+        },
+        ...sharedOptions,
+      ],
+      args: {
+        name: "shell",
+        isOptional: true,
+        suggestions: ["bash", "zsh"],
```
Comment (QuiiBz) [inline dev/hasura.ts:184]: Would love to see `fish` here - more and more peoples are using it!

File: dev/hasura.ts [hunk 3]
```diff
@@ -0,0 +1,732 @@
...
+          name: "--no-browser",
+          description: "Do not automatically open console in browser",
+        },
+        {
+          name: "--static-dir",
+          description:
+            "Directory where static assets mentioned in the console html template can be served from",
+          args: {
+            name: "string",
+            template: "filepaths",
```
Comment (QuiiBz) [inline dev/hasura.ts:237]: Based on the name and description, shouldn't the template be `folders` instead of `filepaths`?

File: dev/hasura.ts [hunk 4]
```diff
@@ -0,0 +1,732 @@
...
+          name: "delete",
+          description: "Clear migrations from local project and server",
+          options: [
+            {
+              name: "--all",
+              description: "Clears all migrations for selected database",
+            },
+            {
+              name: "--force",
+              description: "Executes operation without any confirmation",
```
Comment (QuiiBz) [inline dev/hasura.ts:503]: Adding `isDangerous` could allow users to identify that this option can be dangerous

File: dev/hasura.ts [hunk 5]
```diff
@@ -0,0 +1,732 @@
...
+          options: [
+            {
+              name: "--database-name",
+              description:
+                "Database name for which the current migrations / seeds belong to",
+              args: stringArg,
+            },
+            {
+              name: "--force",
+              description: "Do not ask for confirmation",
```
Comment (QuiiBz) [inline dev/hasura.ts:646]: Same as above

**Review by QuiiBz — CHANGES_REQUESTED** [review]

File: dev/hasura.ts (hunk 2 above)
Comment (Monkeyank) [reply dev/hasura.ts:184]: The only supported completions by asura are `bash` and `zsh`. [Docs Section](https://hasura.io/docs/latest/graphql/core/hasura-cli/hasura_completion.html)

Comment (QuiiBz) [reply dev/hasura.ts:187]: Alright, thanks for the clarification!

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**